reference-implementation/lib/readable-stream-abstract-ops.js

```diff
@@ -159,6 +159,42 @@ export function IsReadableStream(x) {
   return true;
 }
 
+export function PutBackIntoReadableStream(stream, chunk) {
+  if (stream._state === 'closed') {
+    throw new TypeError('stream is closed');
```

Comment:
I think you're right. I'm not sure what the right solution would be though :(. Maybe, after reading the last chunk from the queue, delay for a turn before deciding to close the stream? That gives you a one-turn grace period.